src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,210 @@
+import classNames from 'classnames';
+import React from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink';
+import { SearchLink } from '../SearchLink';
+
+type Props = {
+  people: Person[]
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+  const [searchParams] = useSearchParams();
+
+  const sort = searchParams.get('sort') || '';
+  const queryParam = searchParams.get('query') || '';
+  const centuriesParam = searchParams.getAll('centuries') || [];
+  const sexParam = searchParams.get('sex');
+
+  const sortChangeHandler = (param: string) => {
+    if (searchParams.toString().includes(param)
+      && searchParams.toString().includes('desc')) {
+      return {
+        sort: null,
+        order: null,
+      };
+    }
+
+    if (searchParams.toString().includes(param)) {
+      return {
+        sort: param,
+        order: 'desc',
+      };
+    }
+
+    return {
+      sort: param,
+      order: null,
+    };
+  };
+
+  const sorting = () => {
+    let newPeople = [...people];
+
+    if (searchParams.has('query')) {
+      newPeople = newPeople.filter(
+        person => person.name.toLowerCase().includes(queryParam.toLowerCase())
+        || person.motherName?.toLowerCase().includes(queryParam.toLowerCase())
+        || person.fatherName?.toLowerCase().includes(queryParam.toLowerCase()),
+      );
+    }
+
+    if (searchParams.has('centuries')) {
+      newPeople = newPeople.filter(person => centuriesParam
+        .includes(String(Math.ceil(person.born / 100))));
+    }
+
+    if (searchParams.has('sex')) {
+      newPeople = newPeople.filter(person => person.sex === sexParam);
+    }
+
+    switch (sort) {
+      case 'name':
+      case 'sex':
+        if (searchParams.has('order')) {
+          newPeople.sort((a, b) => b[sort].localeCompare(a[sort]));
+        } else {
+          newPeople.sort((a, b) => a[sort].localeCompare(b[sort]));
+        }
+
+        break;
+
+      case 'born':
+      case 'died':
+        if (searchParams.has('order')) {
+          newPeople.sort((a, b) => b[sort] - a[sort]);
+        } else {
+          newPeople.sort((a, b) => a[sort] - b[sort]);
+        }
+
+        break;
+      default:
+    }
+
+    return newPeople;
+  };
+
+  const getSortIconClass = (sortParam: string) => {
+    return classNames(
+      'fas',
+      {
+        'fa-sort-down': sort === sortParam
+      && searchParams.has('order'),
+      },
+      { 'fa-sort': sort !== sortParam },
+      {
+        'fa-sort-up': sort === sortParam
+          && !searchParams.has('order'),
+      },
+    );
+  };
+
+  const visiblePeople = sorting();
+
+  return (
+    <>
```

Comment:
You can avoid using `Fragment` here. You can just return:
```
return !visiblePeople.length
  ? <p>...</p>
  : (<table>...
```
It's not a mistake, it's just a little improvement.